tests/middleware/test_base.py

```diff
@@ -210,24 +211,17 @@ async def dispatch(
     ) -> Response:
         ctxvar.set("set by middleware")
         resp = await call_next(request)
-        assert ctxvar.get() == "set by endpoint"
+        # BaseHTTPMiddleware creates a TaskGroup which copies the context
+        # and erases any changes to it made within the TaskGroup
+        assert ctxvar.get() == "set by middleware"
```

Comment:
We had the `pytest.mark.xfail` to avoid this comment.

Reply:
Given the comment on  https://github.com/encode/starlette/pull/2777#discussion_r1864925790, can we change the `AssertionError` for another error? `ValueError` maybe?

Reply:
Not quite, any error within a middleware would cause a dangling stream, but since this is addressed in this PR as well, I've reverted the change, so `xfail` in rightfully back now